source/extensions/filters/network/kafka/mesh/request_processor.cc

```diff
@@ -0,0 +1,31 @@
+#include "source/extensions/filters/network/kafka/mesh/request_processor.h"
+
+#include "envoy/common/exception.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace Kafka {
+namespace Mesh {
+
+// Helper function. Throws a nice message. Filter will react by closing the connection.
+static void throwOnUnsupportedRequest(const std::string& reason, const RequestHeader& header) {
+  throw EnvoyException(absl::StrCat(reason, " Kafka request (key=", header.api_key_, ", version=",
+                                    header.api_version_, ", cid=", header.correlation_id_));
+}
+
+void RequestProcessor::onMessage(AbstractRequestSharedPtr arg) {
+  // This will be replaced with switch on header's API key.
+  throwOnUnsupportedRequest("unsupported (bad client API invoked?)", arg->request_header_);
```

Comment:
when `arg` can be actually turned into instance of [InFlightRequest](https://github.com/envoyproxy/envoy/pull/17374/files#diff-bc27f6cc82b6e0abc63f74ed16db37d6b3319fed85d487fecf68d639de5ba287R16) then we can [go back to filter](https://github.com/envoyproxy/envoy/pull/17374/files#diff-54af050994ce6e17b4f69b01159b9ba08721f38cb4698d6ad717cf6b3ee03626R57)